File: Source/Aladdin.CAPI.Environment/ConfigRand.cs
using System;
using System.IO;
using System.Xml;

namespace Aladdin.CAPI.Environment
{
    ///////////////////////////////////////////////////////////////////////////
    // Элемент описания генератора случайных данных
    ///////////////////////////////////////////////////////////////////////////
    public class ConfigRand 
    {
        // имя элемента, имя класса фабрики и признак обязательного использования 
        private string name; private string className; private string critical; 
    
        // конструктор
        public ConfigRand(XmlElement element)
        {
            // получить имя элемента 
            name = element.GetAttribute("name"); 
            // проверить наличие имени элемента
            if (name.Length == 0) throw new IOException(); 
        
            // получить класс фабрики
            className = element.GetAttribute("class"); 
            // проверить наличие класса фабрики
            if (className.Length == 0) throw new IOException(); 

            // получить признак обязательного использования
            critical = element.GetAttribute("critical"); 
        }
        // имя элемента
        public string Name { get { return name; }}
        // класс фабрики
        public string Class { get { return className; }} 
    
        // признак обязательного использования
        public bool Critical { get  
        { 
            // признак обязательного использования
            return (critical.Length > 0) ? Boolean.Parse(critical) : false; 
        }} 
    }
}

File: Source/Aladdin.CAPI.Environment.Java/src/aladdin/capi/environment/ConfigRand.java
package aladdin.capi.environment;
import java.io.*;
import org.w3c.dom.*;

///////////////////////////////////////////////////////////////////////////
// Элемент описания генератора случайных данных
///////////////////////////////////////////////////////////////////////////
public class ConfigRand 
{
    // имя элемента и признак обязательного использования 
    private final String name; private final String critical; 
    // имя модуля и класса фабрики
    private final String classLoader; private final String className; 
    
    // конструктор
    public ConfigRand(Element element) throws IOException
    {
        // получить имя элемента 
        name = element.getAttribute("name"); 
        // проверить наличие имени элемента
        if (name.length() == 0) throw new IOException(); 
        
        // получить имя модуля
        classLoader = element.getAttribute("classLoader"); 
        // проверить наличие имени модуля
        if (classLoader.length() == 0) throw new IOException(); 
        
        // получить класс фабрики
        className = element.getAttribute("className"); 
        // проверить наличие класса фабрики
        if (className.length() == 0) throw new IOException(); 

        // получить признак обязательного использования
        critical = element.getAttribute("critical"); 
    }
    // имя элемента
    public final String name() { return name; } 
    
    // модуль фабрики
    public final String classLoader() { return classLoader; } 
    // класс фабрики
    public final String className() { return className; } 
    
    // признак обязательного использования
    public final boolean critical() 
    { 
        // признак обязательного использования
        return (critical.length() > 0) ? Boolean.parseBoolean(critical) : false; 
    } 
}
